rtengine/rawimagesource.cc

```diff
@@ -3895,1993 +3895,491 @@ void RawImageSource::getRowStartEnd (int x, int &start, int &end)
+                        nh = 112;
+                    } else if (yc[y][x] < 0.45f) {
+                        nh = 113;
+                    } else if (yc[y][x] < 0.5f) {
+                        nh = 114;
+                    } else if (yc[y][x] < 0.55f) {
+                        nh = 115;
+                    } else if (yc[y][x] < 0.6f) {
+                        nh = 116;
+                    }
+                } else if (xc[y][x] < 0.445f && yc[y][x] > 0.15f) {//45
+                    if (yc[y][x] < 0.2f) {
+                        nh = 117;
+                    } else if (yc[y][x] < 0.24f) {
+                        nh = 118;
+                    } else if (yc[y][x] < 0.29f) {
+                        nh = 119;
+                    } else if (yc[y][x] < 0.32f) {
+                        nh = 120;
+                    } else if (yc[y][x] < 0.34f) {
+                        nh = 121;
+                    } else if (yc[y][x] < 0.37f) {
+                        nh = 122;
+                    } else if (yc[y][x] < 0.4f) {
+                        nh = 123;
+                    } else if (yc[y][x] < 0.45f) {
+                        nh = 124;
+                    } else if (yc[y][x] < 0.5f) {
+                        nh = 125;
+                    } else if (yc[y][x] < 0.55f) {
+                        nh = 126;
+                    } else if (yc[y][x] < 0.58f) {
+                        nh = 127;
+                    }
+                } else if (xc[y][x] < 0.495f && yc[y][x] > 0.15f) {
+                    if (yc[y][x] < 0.2f) {
+                        nh = 128;
+                    } else if (yc[y][x] < 0.24f) {
+                        nh = 129;
+                    } else if (yc[y][x] < 0.29f) {
+                        nh = 130;
+                    } else if (yc[y][x] < 0.32f) {
+                        nh = 131;
+                    } else if (yc[y][x] < 0.34f) {
+                        nh = 132;
+                    } else if (yc[y][x] < 0.37f) {
+                        nh = 133;
+                    } else if (yc[y][x] < 0.4f) {
+                        nh = 134;
+                    } else if (yc[y][x] < 0.45f) {
+                        nh = 135;
+                    } else if (yc[y][x] < 0.5f) {
+                        nh = 136;
+                    } else if (yc[y][x] < 0.55f) {
+                        nh = 137;
+                    }
+                } else if (xc[y][x] < 0.545f && yc[y][x] > 0.15f) {
+                    if (yc[y][x] < 0.2f) {
+                        nh = 138;
+                    } else if (yc[y][x] < 0.24f) {
+                        nh = 139;
+                    } else if (yc[y][x] < 0.29f) {
+                        nh = 140;
+                    } else if (yc[y][x] < 0.32f) {
+                        nh = 141;
+                    } else if (yc[y][x] < 0.34f) {
+                        nh = 142;
+                    } else if (yc[y][x] < 0.37f) {
+                     
```

Comment:
`SQR` --> `std::sqrt()`?

Reply:
No idea, why GitHub presents the wrong lines here. Just search for "SQR".

Reply:
SQR squares

Reply:
rt_math.h
```
template<typename T>
constexpr T SQR(T x)
{
    return x * x;
}
```